lgc/patch/Patch.cpp

```diff
@@ -378,6 +371,13 @@ void Patch::addOptimizationPasses(lgc::PassManager &passMgr, CodeGenOpt::Level o
   fpm.addPass(createFunctionToLoopPassAdaptor(std::move(lpm2), true));
   fpm.addPass(LoopUnrollPass(
       LoopUnrollOptions(optLevel).setPeeling(true).setRuntime(false).setUpperBound(false).setPartial(false)));
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 444780
+  // Old version of the code
+  fpm.addPass(SROAPass());
+#else
+  // New version of the code (also handles unknown version, which we treat as latest)
+  fpm.addPass(SROAPass(SROAOptions::ModifyCFG));
```

Comment:
When we were discussing this earlier, I didn't realise that you were proposing moving the SROA rather than adding another one - but this is better.

Reply:
Yes, this could be the reason why average compilation time almost did not changed. I'm planning to add SROA after second LoopUnrollPass, but need to do more testing.